tests/test_learning_role.py

```diff
@@ -25,28 +25,31 @@
 
 @pytest.mark.require_learning
 def test_learning_init():
-    learning_config: LearningConfig = {
+    config = {
         "obs_dim": 3,
         "act_dim": 2,
-        "train_freq": "1h",
         "unique_obs_dim": 0,
-        "algorithm": "matd3",
-        "actor_architecture": "mlp",
-        "learning_mode": False,
-        "evaluation_mode": False,
-        "training_episodes": 3,
-        "episodes_collecting_initial_experience": 1,
-        "continue_learning": False,
-        "trained_policies_save_path": None,
-        "early_stopping_steps": 10,
-        "early_stopping_threshold": 0.05,
+        "learning_config": LearningConfig(
+            train_freq="1h",
+            algorithm="matd3",
+            actor_architecture="mlp",
+            learning_mode=True,
+            evaluation_mode=False,
+            training_episodes=3,
+            episodes_collecting_initial_experience=1,
+            continue_learning=False,
+            trained_policies_save_path=None,
+            early_stopping_steps=10,
+            early_stopping_threshold=0.05,
+        ),
     }
+
     # test init
-    learn = Learning(learning_config, start, end)
+    learn = Learning(config["learning_config"], start=start, end=end)
     assert len(learn.rl_strats) == 0
 
     # we need to add learning strategies first
-    learn.rl_strats["test_id"] = LearningStrategy(**learning_config)
+    learn.rl_strats["test_id"] = LearningStrategy(**config, learning_role=learn)
```

Comment:
```suggestion
    learn.rl_strats["test_id"] = LearningStrategy(**config["learning_config"], learning_role=learn)
```
This should not have the whole config itself passed right?
I would like to ditch this, by removing the unpacked dict:

```suggestion
    learn.rl_strats["test_id"] = LearningStrategy(learning_role=learn)
```

As we have access through `learning_role.learning_config` anyway